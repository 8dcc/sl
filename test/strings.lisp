;;------------------------------------------------------------------------------
;; Features tested in this source:
;;   - String evaluation (single-line and multi-line)
;;   - String predicates: `equal?', `<', `>'
;;------------------------------------------------------------------------------

"Hello, world!"
""

"Multi-line
strings
supported." ; Expected: "Multi-line\nstrings\nsupported."

(concat "Concatenating" " multiple " "strings...")

(equal?
 "All printed strings
must be valid inputs."                         ; Initial string
 "All printed strings\nmust be valid inputs.") ; Printed after evaluating

(substring "--Testing substrings--")
(substring "--Testing substrings--" 2 20)
(substring "--Testing substrings--" 2 -2)
(substring "--Testing substrings--" 10)
(substring "--Testing substrings--" -12)

(< "abc" "abz") ; Expected: tru
(< "abc" "abc") ; Expected: nil
(> "abz" "abc") ; Expected: tru
(> "abc" "abc") ; Expected: nil
