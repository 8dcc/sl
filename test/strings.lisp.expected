"Hello, world!"
"Multi-line\nstrings\nsupported."
0
3
""
"Concatenating multiple strings..."
"(+ 1 2 3 (- 5 4))"
"(lambda (x) (quote symbol) (func x 123 \"Hello, world!\\n\"))"
"%s: Testing format specifiers!"
"%d: 1 2 3"
"%f: 1.000000 2.000000 3.000000"
"--Testing substrings--"
"Testing substrings"
"Testing substrings"
"substrings--"
"substrings--"
<macro>
((0 . 15))
((0 . 15))
((0 . 34) (0 . 7))
nil
((0 . 34) (0 . 30) (31 . 34))
tru
tru
nil
tru
nil
